pkg/controller/daemon/daemoncontroller.go

```diff
@@ -425,11 +425,28 @@ func (dsc *DaemonSetsController) addNode(obj interface{}) {
 	}
 }
 
+func compareConditions(oldConditions, curConditions []v1.NodeCondition) bool {
+	if len(oldConditions) != len(curConditions) {
+		return false
+	}
+	for i := range oldConditions {
```

Comment:
I guessed it was, otherwise the reflect.DeepEqual approach would also be flawed. However, I'm not really sure, from taking a look at nodes in my cluster it seems like it. I could add a sort call before iterating just to be sure.